src/cdk/tree/tree-control.ts

```diff
@@ -0,0 +1,223 @@
+  /** Whether a given node is expanded or not. Returns true if the node is expanded. */
+  expanded(node: T) {
+    return this.expansionModel.isSelected(node);
+  }
+
+  /** Toggles a subtree rooted at `node` recursively. */
+  toggleDecedents(node: T) {
+    this.expansionModel.toggle(node);
+    let expand = this.expansionModel.isSelected(node);
+    expand ? this.expandDecedents(node) : this.collapseDecedents(node);
+  }
+
+  /** Collapse all nodes in the tree. */
+  collapseAll() {
+    this.expansionModel.clear();
+    this.expandChange.next(this.expansionModel.selected);
+  }
+
+  /** Expands all nodes in the tree. */
+  abstract expandAll();
+
+  /** Expands a subtree rooted at given `node` recursively. */
+  abstract expandDecedents(node: T);
+
+  /** Collapses a subtree rooted at given `node` recursively. */
+  abstract collapseDecedents(node: T);
+
+  /** Gets a list of decedent nodes of a subtree rooted at given `node` recursively. */
+  abstract getDecedents(node: T): T[];
+}
+
+/** Flat tree control. Able to expand/collapse a subtree recursively for FlatNode type. */
+export class FlatTreeControl<T extends FlatNode> extends BaseTreeControl<T> {
+  /**
+   * Gets a list of decedent nodes of a subtree rooted at given `node` recursively.
+   *
+   * To make this working, the `nodes` of the TreeControl must be set correctly.
+   */
+  getDecedents(node: T) {
+    const startIndex = this.nodes.indexOf(node);
+    const results: T[] = [];
+    for (let i = startIndex + 1; i < this.nodes.length && node.level < this.nodes[i].level; i++) {
+      results.push(this.nodes[i]);
+    }
+    return results;
+  }
+
+  /**
+   * Expands all nodes in the tree.
+   *
+   * To make this working, the `nodes` of the TreeControl must be set correctly.
+   */
+  expandAll() {
+    this.expansionModel.clear();
+    this.nodes.forEach((node) => node.expandable && this.expansionModel.select(node));
+    this.expandChange.next(this.expansionModel.selected);
+  }
+
+  /** Expands a subtree rooted at given `node` recursively. */
+  expandDecedents(node: T) {
+    let decedents = this.getDecedents(node);
+    decedents.forEach((child) => child.expandable && this.expansionModel.select(child));
+    this.expandChange.next(this.expansionModel.selected);
+  }
+
+  /** Collapses a subtree rooted at given `node` recursively. */
+  collapseDecedents(node: T) {
+    let decedents = this.getDecedents(node);
+    decedents.forEach((child) => this.expansionModel.deselect(child));
+    this.expandChange.next(this.expansionModel.selected);
+  }
+}
+
+/** Nested tree control. Able to expand/collapse a subtree recursively for NestedNode type. */
+export class NestedTreeControl<T extends NestedNode> extends BaseTreeControl<T> {
+  /**
+   * Expands all nodes in the tree.
+   *
+   * To make this working, the `nodes` of the TreeControl must be set correctly.
+   */
+  expandAll() {
+    this.expansionModel.clear();
+    this.nodes.forEach((node) => this._expandDecedents(node));
```

Comment:
Nit: don't need parens for a single argument with inferred type (here and elsewhere)